src/cli/glue_gen.rs

```diff
@@ -27,22 +27,39 @@ impl GlueCrateGenerator {
         self.add_glue_code(temp_path.as_path())
             .context("Failed to add glue code")?;
 
+        // create a gluecrate with the temp_path and that can access the manifest path
+
         todo!()
     }
 
     fn create_temp_dir(&self) -> AnyResult<PathBuf> {
+
+        // Use tempdir to create a temporary directory for the current and previous api 
+        // that should drop at the end of the comparison
+
         todo!()
     }
 
     fn copy_current_version(&self, glue_path: &Path) -> AnyResult<()> {
+        // Cargo new a current folder in the tempdir path
+        // cargo edit its dependencies
+        // copy the code inside
+        // build it
+
```

Comment:
Maybe we can do the Rust-equivalent of `cp ./ $GLUE_PATH/current` that you would type in Bash. AFAIK, nothing more is needed.

While this is good enough for the first implementation, it would be a good idea not to copy the build artifacts in the `target` folder, and basically anything that is `gitignore`d. I think this can be done with [`Repository::checkout_index`], but this is overkill for now.

[`Repository::checkout_index`]: https://docs.rs/git2/0.13.20/git2/struct.Repository.html#method.checkout_index